python/taichi/lang/impl.py

```diff
@@ -832,6 +838,10 @@ def vars2entries(_vars):
                 if len(_var) > 0 and isinstance(
                         _var[0], str) and _var[0] == '__ti_format__':
                     res = _var[1:]
+                elif len(_var) > 0 and isinstance(
+                        _var[0], str) and _var[0] == '__ti_fmt_list__':
```

Comment:
I was thinking `__ti_fmt_list__` functions the same way as `__ti_format__`? What's this attribute used for?

Reply:
These are less relevant code for handling formats extraction from the `str.format()` method. I have removed them in this PR to focus on f-strings.

Reply:
I rewrite this part of code and added a tag `'__ti_fmt_value__'` when building `FormattedValue` (the Python AST counterpart of an expr-format pair `{x:.2f}` in the f-string).

In this way, the Formatted value is converted to the form `['__ti_fmt_value__', Expr, format]` and can therefore be distinguished from a normal list to be printed containing two elements. Thus in `ti_format()` we can use another code path to yield this pair and prevent it being interpreted as a normal list.